compal/__init__.py

```diff
@@ -926,6 +927,204 @@ def turn_off(self, debug=False):
+                        changes[attr],
+                    )
+                else:
+                    new_value = getattr(new_settings, attr)
+                    if old_value != new_value:
+                        changes.update({attr: f"{old_value} -> {new_value}"})
+                        bln_changes[0] = True
+
+        changes = {}
+        bln_changes = [False]
+        iterate_changes(old_settings.guest_networks_2g[interface_index],
+                        new_settings.guest_networks_2g[interface_index], changes)
+        iterate_changes(old_settings.guest_networks_5g[interface_index],
+                        new_settings.guest_networks_5g[interface_index], changes)
+        return bln_changes[0], changes
+
+    def __check_router_status(self, new_guest_network_settings, interface_index, debug):
+        """
+        Checks if (1) the new user settings were changed in the router (via checking
+        the getter fun:307 [WIRELESSGUESTNETWORK] in the router-xml-file), (2) prints out a
+        progress bar and (3) prints out if the process was successful.
+        """
+        router_settings = None
+
+        progress = 0
+
+        def print_progress_bar(progress, total, postfix=""):
+            progress = total if progress > total else progress
+            per_progress = int(progress / total * 100)
+            postfix = f"[{postfix}]" if postfix != "" else ""
+            print(
+                f"\r|{'█' * progress}{'-' * (total - progress)}| {per_progress}%, {progress}sec\t{postfix}",
+                end="",
+            )
+
+        total = 24
+        bln_changes = True
+        start_time = time.time()
+        changes = ""
+        if debug:
+            print(f"\n--- WAITING FOR ROUTER TO UPDATE ---")
+        while progress < total and bln_changes is True:
+            time.sleep(3)
+            if debug:
+                print_progress_bar(progress, total, changes)
+            try:
+                router_settings = self.wifi_guest_network_settings
+                bln_changes, changes = self.__compare_wifi_settings(
+                    router_settings, new_guest_network_settings, interface_index, changes
+                )
+            except Exception as e:
+                changes = str(e)
+        if debug:
+            if not bln_changes:
+                print(f"\n\n--- ROUTER SUCESSFULLY UPDATED ALL NEW WIFI SETTINGS! ---")
+            else:
+                print(f"\n\n--- CHANGES THAT DID NOT GET SET ---")
+                _, changes = self.__compare_wifi_settings(router_settings, new_guest_network_settings, interface_index)
+                print(changes)
+        return bln_changes
+
+    def update_interface_guest_network_settings(self, new_guest_network_settings, interface_index, debug=True):
+        """
+        Method for updating the wifi guest network settings. Uses fun:308.
+        """
+        assert 0 <= interface_index < (len(new_guest_network_settings.guest_networks_2g) and
```

Comment:
I think `(len(new_guest_network_settings.guest_networks_2g) and                                        len(new_guest_network_settings.guest_networks_5g))` probably does not do what you expect. It returns 0 if the first value is 0, otherwise it returns the second value.

I would write:
```
assert 0 <= interface_index and interface_index < max(len(new_guest_network_settings.guest_networks_2g), len(new_guest_network_settings.guest_networks_5g))
```

But please make sure that does what you intend

Reply:
Oh no, what a pity, I must have been tired when I wrote this ;-)  
You are right, it was not what I intended to do.
It is fixed now.